pkg/virt-operator/resource/apply/core_test.go

```diff
@@ -88,6 +94,174 @@ var _ = Describe("Apply", func() {
+			stores = util.Stores{}
+			stores.ConfigMapCache = cache.NewStore(cache.DeletionHandlingMetaNamespaceKeyFunc)
+			stores.InstallStrategyConfigMapCache = cache.NewStore(cache.MetaNamespaceKeyFunc)
+
+			expectations = &util.Expectations{}
+
+			clientset = kubecli.NewMockKubevirtClient(ctrl)
+			clientset.EXPECT().KubeVirt(Namespace).Return(kvInterface).AnyTimes()
+			clientset.EXPECT().CoreV1().Return(coreclientset.CoreV1()).AnyTimes()
+
+			kv = &v1.KubeVirt{}
+		})
+
+		AfterEach(func() {
+			ctrl.Finish()
+		})
+
+		It("should not patch ConfigMap on sync", func() {
+			requiredCM := components.NewKubeVirtCAConfigMap(operatorNamespace)
+			version, imageRegistry, id := getTargetVersionRegistryID(kv)
+			injectOperatorMetadata(kv, &requiredCM.ObjectMeta, version, imageRegistry, id, true)
+
+			existingCM := requiredCM.DeepCopy()
+			crt := createCrt()
+
+			bundle, _, err := components.MergeCABundle(crt, []byte(cert.EncodeCertPEM(crt.Leaf)), time.Hour)
+			Expect(err).ToNot(HaveOccurred())
+
+			existingCM.Data = map[string]string{
+				components.CABundleKey: string(bundle),
+			}
+
+			stores.ConfigMapCache.Add(existingCM)
+
+			r := &Reconciler{
+				kv:           kv,
+				stores:       stores,
+				clientset:    clientset,
+				expectations: expectations,
+			}
+
+			_, err = r.createOrUpdateKubeVirtCAConfigMap(queue, crt, duration, requiredCM)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should patch ConfigMap on sync when not parsable", func() {
+			notRSAParsableString := "something not parsable"
+			requiredCM := components.NewKubeVirtCAConfigMap(operatorNamespace)
+			version, imageRegistry, id := getTargetVersionRegistryID(kv)
+			injectOperatorMetadata(kv, &requiredCM.ObjectMeta, version, imageRegistry, id, true)
+
+			existingCM := requiredCM.DeepCopy()
+			existingCM.Data = map[string]string{
+				components.CABundleKey: notRSAParsableString,
+			}
+			stores.ConfigMapCache.Add(existingCM)
+
+			r := &Reconciler{
+				kv:           kv,
+				stores:       stores,
+				clientset:    clientset,
+				expectations: expectations,
+			}
+
+			patched := false
+			coreclientset.Fake.PrependReactor("patch", "configmaps", func(action testing.Action) (handled bool, ret runtime.Object, err error) {
+				a := action.(testing.PatchActionImpl)
+				patch, err := jsonpatch.DecodePatch(a.Patch)
+				Expect(err).ToNot(HaveOccurred())
+
+				obj, err := json.Marshal(existingCM)
+				Expect(err).To(BeNil())
+
+				obj, err = patch.Apply(obj)
+				Expect(err).To(BeNil())
+
+				pr := &corev1.ConfigMap{}
+				Expect(json.Unmarshal(obj, existingCM)).To(Succeed())
+				Expect(existingCM.Data[components.CABundleKey]).ToNot(Equal(notRSAParsableString))
+
+				patched = true
+				return true, pr, nil
+			})
+
+			crt := createCrt()
+
+			_, err := r.createOrUpdateKubeVirtCAConfigMap(queue, crt, duration, requiredCM)
+			Expect(err).To(BeNil())
+			Expect(patched).To(BeTrue())
+		})
+
+		It("should patch ConfigMap on sync when CA expired", func() {
```

Comment:
@ashleyschuett Hi. Just wanted to know if we actually patch because we hit the renewBefore(overlap) interval? during that interval we will hold both old and new CA, and when that interval ends we hit another patch to remove the old one, am I correct?

Reply:
yes correct

Reply:
@ashleyschuett Sorry for getting stuck on this, but sometimes it is clear how the feature behaves when you read the unit/e2e tests. I think it is better to update the test description to something like `should patch CA ConfigMap on sync when  RenewBefore period is reached`

Reply:
I don't think that's entirely accurate. This configmap sync does not decide or know the renewbefore info. Updating the CA happens before getting to configmap sync when checking the secret that holds the CA. In this function we only know what the CA passed in is and if the bundle we create from the CA matches what we currently have. It's not about the RenewBefore period its just checking bundleFromCurrentCA != bundleSavedInConfigMap.

Reply:
@ashleyschuett Thank you for elaborating on this, indeed now it seems more accurate :)